daft/daft/__init__.pyi

```diff
@@ -1184,7 +1184,9 @@ class SQLFunctionStub:
     def arg_names(self) -> list[str]: ...
 
 def sql_exec(source: str, session: PySession, config: PyDaftPlanningConfig) -> object | None: ...
-def sql(sql: str, catalog: PyCatalog, daft_planning_config: PyDaftPlanningConfig) -> LogicalPlanBuilder: ...
+def sql(
+    sql: str, catalog: PyCatalog, session: PySession, daft_planning_config: PyDaftPlanningConfig
+) -> LogicalPlanBuilder: ...
```

Comment:
Perhaps we could use sql_exec now. If the session is going to be cloned, we can attach the provided catalog (if any) from daft.sql then call sql_exec.

Reply:
it seems like it'd be the same thing but just moving the logic in to python. 

```py
    _py_catalog = catalog._catalog

    sess = daft.current_session()._session
    cloned_sess = # clone session somehow

    _py_logical = _sql(sql, _py_catalog, cloned_sess, planning_config)
```

Reply:
We want to call `sql_exec` since it has the logic to execute session statements which is why it takes the session. So yes we want to move the logic in to python so we can call the appropriate rust function.

Reply:
so it's actually a lot of work to move the cloning behavior in to python, and it doesn't  add any value. Will keep this as is with the separate method for now, but use the same logic as `sql_exec`. Once we completely get rid of `SQLCatalog` we can merge them in to one.